patches/chromium/gin_dictionary_default_constructor.patch

```diff
@@ -0,0 +1,37 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Cheng Zhao <zcbenz@gmail.com>
+Date: Thu, 4 Oct 2018 14:57:02 -0700
+Subject: gin_dictionary_default_constructor.patch
+
+Add default constructor for gin::Dictionary.
+
+This is required for automatically converting arguments for functions that
+take gin::Dictionary as parameter.
+
+diff --git a/gin/dictionary.cc b/gin/dictionary.cc
+index 95e00072700c..7643347890a5 100644
+--- a/gin/dictionary.cc
++++ b/gin/dictionary.cc
+@@ -6,6 +6,10 @@
+ 
+ namespace gin {
+ 
++Dictionary::Dictionary()
```

Comment:
For such call sites, can we resort to having `gin::Arguments` as the argument and extract `gin::Dictionary` from that in the callback, which would avoid this patch. ex:

```
void AtomDownloadManagerDelegate::OnDownloadSaveDialogDone(..., gin::Arguments* args) {
  gin::Dictionary result(args->isolate());
  // extract result from args.
}
```


Ideally it would be better if we rewrite `ArgumentsHolder` to not expect a default constructor.

Reply:
As an alternative (because I don't like the idea of duplicate that dictionary construction code everywhere).  Can we just extend `gin::Dictionary` into `electron::util::Dictionary` and implement the empty constructor there?

Reply:
We already have an extended `gin_helper::Dictionary` and it should be possible to use it to avoid patching `gin`. I'll put it in the next PR.